packages/astro/src/core/build/internal.ts

```diff
@@ -189,6 +202,31 @@ export function* eachPageData(internals: BuildInternals) {
 	yield* internals.pagesByComponent.values();
 }
 
+export function hasPrerenderedPages(internals: BuildInternals) {
+	for (const id of internals.pagesByViteID.keys()) {
+		if (internals.pageOptionsByPage.get(id)?.prerender) {
+			return true
+		}
+	}
+	return false
+}
+
+export function* eachPrerenderedPageData(internals: BuildInternals) {
+	for (const [id, pageData] of internals.pagesByViteID.entries()) {
+		if (internals.pageOptionsByPage.get(id)?.prerender) {
+			yield pageData;
+		}
+	}
+}
+
+export function* eachServerPageData(internals: BuildInternals) {
+	for (const [id, pageData] of internals.pagesByViteID.entries()) {
+		if (!internals.pageOptionsByPage.get(id)?.prerender) {
+			yield pageData;
+		}
+	}
+}
+
```

Comment:
The Build API gets two new utilities which map over prerendered pages and server-only pages.